src/avro/src/reader.rs

```diff
@@ -556,24 +556,21 @@ impl<'a> SchemaResolver<'a> {
                 SchemaPieceRefOrNamed::Piece(SchemaPiece::Union(w_inner)),
                 SchemaPieceRefOrNamed::Piece(SchemaPiece::Union(r_inner)),
             ) => {
-                // TODO (brennan) - can we avoid cloning this?
                 let w2r = self.writer_to_reader_names.clone();
                 let permutation = w_inner
```

Comment:
I know that this permutation is described in the comment above, but possibly having a simple comment like `// Some(val) iff piece is resolvable in both reader and writer, else None` would help me understand this logic.